src/utils/KML.js

```diff
@@ -277,36 +297,36 @@ const readFeatures = (kmlString, featureProjection) => {
  * @param {VectorLayer} layer A react-spatial VectorLayer.
  * @param {<ol.Projection|String>} featureProjection The current projection used by the features.
  */
-const writeFeatures = (layer, featureProjection) => {
+const writeFeatures = (layer, featureProjection, mapResolution) => {
   let featString;
   const { olLayer } = layer;
   const exportFeatures = [];
 
-  olLayer.getSource().forEachFeature((f) => {
+  olLayer.getSource().forEachFeature((feature) => {
     // We silently ignore Circle elements as they are
     // not supported in kml.
-    if (f.getGeometry().getType() === 'Circle') {
+    if (feature.getGeometry().getType() === 'Circle') {
       return;
     }
 
-    const clone = f.clone();
-    clone.setId(f.getId());
-    clone.getGeometry().setProperties(f.getGeometry().getProperties());
+    const clone = feature.clone();
+    clone.setId(feature.getId());
+    clone.getGeometry().setProperties(feature.getGeometry().getProperties());
     clone.getGeometry().transform(featureProjection, 'EPSG:4326');
 
     // We remove all ExtendedData not related to style.
-    Object.keys(f.getProperties()).forEach((key) => {
+    Object.keys(feature.getProperties()).forEach((key) => {
       if (!/^(geometry|name|description)$/.test(key)) {
         clone.unset(key, true);
       }
     });
 
     let styles;
 
-    if (clone.getStyleFunction()) {
-      styles = clone.getStyleFunction()(clone);
+    if (feature.getStyleFunction()) {
+      styles = feature.getStyleFunction()(feature, mapResolution);
     } else if (olLayer && olLayer.getStyleFunction()) {
-      styles = olLayer.getStyleFunction()(clone);
+      styles = olLayer.getStyleFunction()(feature, mapResolution);
     }
```

Comment:
can you manage the fact that a styleFunction return only one style an dont an array of style, pls?

Reply:
Done